### PR TITLE
Add rooms to Calling sample

### PR DESCRIPTION
**Pull request by mgamis-msft**
# What

Add rooms to Calling sample

# Why

https://skype.visualstudio.com/SPOOL/_workitems/edit/2940134

# How Tested

Tested sample on beta and stable

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: samples/Calling/src/app/views/HomeScreen.tsx [hunk 1]
```diff
@@ -165,6 +221,8 @@ export const HomeScreen = (props: HomeScreenProps): JSX.Element => {
                 props.startCallHandler({
                   displayName,
                   teamsLink,
+                  /* @conditional-compile-remove(rooms) */
```
Comment (alkwa-msft) [inline samples/Calling/src/app/views/HomeScreen.tsx:224]: this startCallhandler. How I am reading this. It means we can supply multiple different locators?I think we only want to ever have one locator?

File: samples/Calling/src/app/utils/AppUtils.ts [hunk 2]
```diff
@@ -37,6 +41,43 @@ export const getGroupIdFromUrl = (): GroupLocator | undefined => {
...
+/* @conditional-compile-remove(rooms) */
+export const joinRoom = async (userId: string, roomId: string, role: Role): Promise<void> => {
+  const requestOptions = {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify({ acsUserId: userId, roomId: roomId, role: role })
+  };
+  const response = await fetch('http://localhost:7071/api/Rooms-AddParticipants', requestOptions);
```
Comment (alkwa-msft) [inline samples/Calling/src/app/utils/AppUtils.ts:75]: can we export this localhost:7071 url to a common variable? I think in the future we are going to ask all backend functionality coming from a target url?

File: samples/Calling/src/app/App.tsx [hunk 3]
```diff
@@ -1,8 +1,9 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-import { GroupCallLocator, TeamsMeetingLinkLocator } from '@azure/communication-calling';
```
Comment (prabhjot-msft) [inline samples/Calling/src/app/App.tsx:4]: Don't we need `GroupCallLocator` & `TeamsMeetingLinkLocator` in the app?

Comment (mgamis-msft) [reply samples/Calling/src/app/App.tsx:4]: Yes, we need for the app. But not for this file. GroupCallLocator & TeamsMeetingLinkLocator are in the union type CallAdapterLocator.

**Review by alkwa-msft — APPROVED** [review]

**Review by prabhjot-msft — APPROVED** [review]

**Pull request merged**